graveler/committed/tree/diff_test.go

```diff
@@ -0,0 +1,350 @@
+package tree_test
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/hex"
+	"errors"
+	"testing"
+
+	"github.com/go-test/deep"
+
+	"github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/graveler/committed"
+	"github.com/treeverse/lakefs/graveler/committed/tree"
+	"github.com/treeverse/lakefs/graveler/testutil"
+)
+
+func TestDiff(t *testing.T) {
+	const (
+		added   = graveler.DiffTypeAdded
+		removed = graveler.DiffTypeRemoved
+		changed = graveler.DiffTypeChanged
+	)
+	tests := map[string]struct {
+		leftKeys                 [][]string
+		leftIdentities           [][]string
+		rightKeys                [][]string
+		rightIdentities          [][]string
+		expectedDiffKeys         []string
+		expectedDiffTypes        []graveler.DiffType
+		expectedDiffIdentities   []string
+		expectedLeftReadsByPart  []int
+		expectedRightReadsByPart []int
```

Comment:
not too much? we just need to check the Diff output

Reply:
Synced f2f: the diff iterator operates on two tree iterators. We want to make sure that it doesn't perform unnecessary operations on these iterators. Namely, we want to verify that identical parts are indeed skipped.